File: Data Structures & Algorithms/longest-consecutive-sequence/submission-1.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        s = set()
        for num in nums:
            s.add(num)

        maxVal = 0
        
        for i in range(len(nums)):
            currNum = nums[i]
            if currNum - 1 in s:
                continue
            temp = 1
            while currNum + 1 in s:
                currNum += 1
                temp += 1
            
            maxVal = max(temp, maxVal)
        
        return maxVal

File: Data Structures & Algorithms/permutations/submission-1.java
// Time complexity: O(n!)
// Space complexity: O(n!)

class Solution {
    public List<List<Integer>> permute(int[] nums) {
        List<List<Integer>> perms = new ArrayList<>();
        perms.add(new ArrayList<>());

        for (int num : nums) {
            List<List<Integer>> temp = new ArrayList<>();
            for (List<Integer> perm : perms) {
                for (int i = 0; i <= perm.size(); i++) {
                    List<Integer> newCopy = new ArrayList<>(perm);
                    newCopy.add(i, num); // Insert at every index
                    temp.add(newCopy);
                }
            }
            perms = temp;
        }

        return perms;
    }
}
